website/src/components/SearchPage/DownloadDialog/LinkOutMenu.tsx

```diff
@@ -0,0 +1,156 @@
+import { Menu, MenuButton, MenuItem, MenuItems } from '@headlessui/react';
+import { type FC, useState, useRef } from 'react';
+
+import { type DownloadUrlGenerator, type DownloadOption } from './DownloadUrlGenerator';
+import { type SequenceFilter } from './SequenceFilters';
+import { processTemplate } from '../../../utils/templateProcessor';
+import BasicModal from '../../common/Modal';
+import DashiconsExternal from '~icons/dashicons/external';
+import IwwaArrowDown from '~icons/iwwa/arrow-down';
+
+const DATA_TYPES = ['unalignedNucleotideSequences', 'metadata', 'alignedNucleotideSequences'] as const;
+type DataType = (typeof DATA_TYPES)[number];
+
+type LinkOut = {
+    name: string;
+    url: string;
+};
+
+type LinkOutMenuProps = {
+    downloadUrlGenerator: DownloadUrlGenerator;
+    sequenceFilter: SequenceFilter;
+    linkOuts: LinkOut[];
+};
+
+export const LinkOutMenu: FC<LinkOutMenuProps> = ({ downloadUrlGenerator, sequenceFilter, linkOuts }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [isModalVisible, setModalVisible] = useState(false);
+    const currentLinkOut = useRef<LinkOut | null>(null);
+
+    const handleLinkClick = (linkOut: LinkOut) => {
+        currentLinkOut.current = linkOut;
+        setModalVisible(true);
+    };
+
+    const generateLinkOutUrl = (linkOut: LinkOut, includeRestricted = false) => {
```

Comment:
Like the comment above regarding `processTemplate()`, I think that it would be great to unit test this due to the complexity of the logic. I could imagine either as a whole or to extract the placeholder matching logic into a separate function and unit test that function. This function can then be simplified to something like `processTemplate(linkOut.url, matchPlaceholders(linkOut.url))`.